pkg/controller/signup.go

```diff
@@ -46,6 +47,80 @@ func (s *Signup) PostHandler(ctx *gin.Context) {
 	ctx.Writer.WriteHeaderNow()
 }
 
+// UpdateVerificationHandler starts the verification process and updates a usersignup resource
+func (s *Signup) UpdateVerificationHandler(ctx *gin.Context) {
+	userID := ctx.GetString(context.SubKey)
+	signup, err := s.signupService.GetUserSignup(userID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			errors.AbortWithError(ctx, http.StatusNotFound, err, "usersignup not found")
+		}
+		log.Error(ctx, err, "error retrieving usersignup")
+		errors.AbortWithError(ctx, http.StatusInternalServerError, err, fmt.Sprintf("error retrieving usersignup: %s", userID))
+		return
+	}
+
+	// check that verification is required before proceeding
+	if signup.Spec.VerificationRequired == false {
+		log.Errorf(ctx, nil, "phone verification not required for usersignup: %s", userID)
+		ctx.AbortWithStatus(http.StatusBadRequest)
+	}
+
+	// Read the Body content
+	var bodyBytes []byte
+	defer func() {
+		if err := ctx.Request.Body.Close(); err != nil {
+			log.Error(nil, err, "failed to close response after reading")
+			return
+		}
+	}()
+
+	if ctx.Request.Body != nil {
+		bodyBytes, err = ioutil.ReadAll(ctx.Request.Body)
+		if err != nil {
+			ctx.AbortWithError(http.StatusInternalServerError, err)
+			return
+		}
+	}
+
+	m := make(map[string]string)
+	err = json.Unmarshal(bodyBytes, &m)
+	if err != nil {
+		log.Errorf(ctx, nil, "Request body could not be read")
+		ctx.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	countryCode := m["country_code"]
+	phoneNumber := m["phone_number"]
+	err = s.signupService.CheckPhoneNumberValid(userID, countryCode, phoneNumber)
+	if err != nil {
+		if apierrors.IsInternalError(err) {
```

Comment:
I would rather check apierrors.ForbiddenError and return http.StatusForbiden. In the case of any other error return StatusInternalServerError.

Reply:
ohhh good idea, updated as part of: a7be0d46781f1feea1afdfca64e727a883b05d47